Closes AgileVentures/MetPlus_tracker#127

patmbolger opened the pull request:
Added agencies_controller with two actions: edit and update
Added spec controller tests
Added cucumber integration tests

joaopapereira commented on app/helpers/agency_admin_helper.rb:
```diff
@@ -4,4 +4,8 @@ def current_user_agency_admin?
     person.agency_roles.pluck(:role).include? AgencyRole::ROLE[:AA]
   end
+  
+  def person_full_name(person)
```
Shouldn't this be a model method?

joaopapereira commented on app/models/agency.rb:
```diff
@@ -9,10 +9,7 @@ class Agency < ActiveRecord::Base
   validates :phone, :phone => true
   validates :email, :email => true
   validates :website, :website => true
-  validates :fax, :phone => true
```
Why do we stop validating the fax?

joaopapereira commented on config/routes.rb:
```diff
@@ -1,24 +1,22 @@
+  
+  root 'main#index'
+  
+  get 'agency_admin/home'
```
Maybe this URL should look like /admin/agency/
Like this is more rest like

joaopapereira commented:
Looks good. Just clarify my comments

patmbolger commented:
I moved the 'full_name' method to User as this will be needed by both AgencyPerson and CompanyPerson.

I moved the agencies controller under the 'admin' namespace.

joaopapereira commented on app/models/user.rb:
```diff
@@ -37,5 +37,9 @@ def self.is_company_contact?(user)
       user.actable.company_roles.pluck(:role).include? CompanyRole::ROLE[:EC]
     end
+    
+    def self.full_name(person)
```
This should be a instance method not a class method

patmbolger replied:
what was I thinking?  thanks for catching that ....

joaopapereira commented on config/routes.rb:
```diff
@@ -1,24 +1,22 @@
+  
+  root 'main#index'
+  
+  get 'agency_admin/home'
```
Shouldn't this rout be under the resources part above?

patmbolger replied:
I moved it under admin.

joaopapereira commented on app/models/agency.rb:
```diff
@@ -9,10 +9,7 @@ class Agency < ActiveRecord::Base
   validates :website, :website => true
-  validates :fax, :phone => true
-  
-  @@agency_admin = nil
```
I believe we have one Issue for these changes, One to refactor this part?
Or are you thinking in the refactor ticket to refector the find_user_with_role function?

patmbolger replied:
Yes, I have refactored that in the next (pending ) PR.

joaopapereira commented:
Looks good.
